driver/src/main/java/org/neo4j/driver/internal/NetworkSession.java

```diff
@@ -174,10 +175,14 @@ public synchronized Transaction beginTransaction()
         return beginTransaction( mode );
     }
 
+    @Deprecated
     @Override
     public synchronized Transaction beginTransaction( String bookmark )
     {
-        lastBookmark = bookmark;
+        if ( bookmark != null )
```

Comment:
How about move the check of null into `setBookmark` method and invoke the method from here? So that all the place where we want to set bookmark then we will just call a single `setBookmark` method.